Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -269,8 +269,8 @@ sendReferralFCM ride booking mbRiderDetails transporterConfig = do
     fraudChecksForReferralPayout mobileNumberHash riderDetails mbDailyStats = do
       availablePersonWithNumber <- SQP.findAllMerchantIdByPhoneNo riderDetails.mobileCountryCode mobileNumberHash
       let isValidForMinPickupThreshold = maybe True (>= transporterConfig.minPickupDistanceThresholdForReferralPayout) booking.distanceToPickup
-          isValidForMinRideDistance = ride.traveledDistance >= transporterConfig.minRideDistanceThresholdForReferralPayout
-          isMaxReferralExceeded = maybe False ((< transporterConfig.maxPayoutReferralForADay) . (.activatedValidRides)) mbDailyStats
+          isValidForMinRideDistance = (fromMaybe (highPrecMetersToMeters ride.traveledDistance) ride.chargeableDistance) >= highPrecMetersToMeters transporterConfig.minRideDistanceThresholdForReferralPayout
```

Comment:
We should consider ride.traveledDistance as source of truth for travel distance. Why did you change this logic?